sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/externalrules/CxxExternalRulesSensor.java

```diff
@@ -87,4 +119,48 @@ public void stream(SMHierarchicCursor rootCursor) throws XMLStreamException {
 
     parser.parse(report);
   }
+
+  private void transformFiles() throws TransformerException {
+    for(int i = 1; i < 100; i++) {
+      String stylesheetKey = SONAR_CXX_XSLT_KEY + i + STYLESHEET_KEY;
+      String sourceKey = SONAR_CXX_XSLT_KEY + i + SOURCE_KEY;
+      String outputKey = SONAR_CXX_XSLT_KEY + i + OUTPUT_KEY;
+
+      String stylesheet = FilenameUtils.normalize(settings.getString(SONAR_CXX_XSLT_KEY + i + STYLESHEET_KEY));
+      String[] sources = settings.getStringArray(SONAR_CXX_XSLT_KEY + i + SOURCE_KEY);
+      String[] outputs = settings.getStringArray(SONAR_CXX_XSLT_KEY + i + OUTPUT_KEY);
+
+      if (sources.length != outputs.length) {
+        LOG.error("Number of source XML files is not equal to the the number of output files.");
+      } else if ((stylesheet != null) ||
+        ((sources != null) && (sources.length > 0)) ||
+        ((outputs != null) && (outputs.length > 0))) {
+        if (stylesheet == null) {
+          LOG.error(SONAR_CXX_XSLT_KEY + i + STYLESHEET_KEY + " is not defined.");
+        } else if (sources == null) {
+          LOG.error(SONAR_CXX_XSLT_KEY + i + SOURCE_KEY + " file is not defined.");
+        } else if (outputs == null) {
+          LOG.error(SONAR_CXX_XSLT_KEY + i + OUTPUT_KEY + " is not defined.");
+        } else {
+          LOG.debug("Converting " + sourceKey + " with " + stylesheetKey + " to " + outputKey + ".");
+          File stylesheetFile = new File(stylesheet);
+          if (stylesheetFile.isAbsolute()) {
+            for (int j = 0; j < sources.length; j++) {
+              transformFile(stylesheetFile, sources[j], outputs[j]);
+            }
+          }
+        }
+      }
+    }
+  }
+
+  private void transformFile(File stylesheetFile, String source, String output) throws TransformerException {
```

Comment:
transformation can be done in memory, as the xunit sensor. it should speed up the analysis